Strata/DDM/Elab.lean

```diff
@@ -407,4 +408,27 @@ def elabDialect
   | .dialect loc dialect =>
     elabDialectRest fm dialects #[] inputContext loc dialect startPos stopPos
 
+def parseDialectIntoConcreteAst (filePath : String) (dialect: Dialect) : IO (InputContext × Strata.Program) := do
+  let dialects := Elab.LoadedDialects.ofDialects! #[initDialect, dialect]
+
+  let bytes ← Strata.Util.readBinInputSource filePath
+  let fileContent ← match String.fromUTF8? bytes with
+    | some s => pure s
+    | none => throw (IO.userError s!"File {filePath} contains non UTF-8 data")
+
+  -- Add program header to the content
+  let contents := s!"program {dialect.name};\n\n" ++ fileContent
+
+  let leanEnv ← Lean.mkEmptyEnvironment 0
+  let inputContext := Strata.Parser.stringInputContext filePath contents
+  let returnedInputContext := {inputContext with
+    fileMap := { source := fileContent, positions := inputContext.fileMap.positions.drop 2 }
+  }
+  let strataProgram ← match Strata.Elab.elabProgram dialects leanEnv inputContext with
```

Comment:
Instead of prepending to context, you could call `elabProgramRest`.  This takes the dialect as an argument.  The loc value could be `default` and startPos be `0`.

Reply:
I'd like that, but trying it I get `Parse errors: 1:0: expected token` and that is hard to debug. Could it tell me exactly what it found and what it is expected? The file starts with a C-style comment, but I was under the impression that DDM dialects support those by default.